tests/nosetests/pyanaconda_tests/module_storage_test.py

```diff
@@ -1039,6 +1059,81 @@ def format_test(self, blockdev):
         ])
 
 
+class FCOEInterfaceTestCase(unittest.TestCase):
+    """Test DBus interface of the FCoE module."""
+
+    def setUp(self):
+        """Set up the module."""
+        self.fcoe_module = FCOEModule()
+        self.fcoe_interface = FCOEInterface(self.fcoe_module)
+
+    def get_nics_test(self):
+        """Test the get nics method."""
+        self.assertEqual(self.fcoe_interface.GetNics(), list())
+
+    def get_dracut_arguments(self):
+        """Test the get dracut arguments method."""
+        self.assertEqual(self.fcoe_interface.GetDracutArguments("eth0"), list())
+
+    @patch('pyanaconda.dbus.DBus.publish_object')
+    def discover_with_task_test(self, publisher):
+        """Test the discover task."""
+        task_path = self.fcoe_interface.DiscoverWithTask(
+            "eth0",
+            False,
+            True
+        )
+
+        publisher.assert_called_once()
+        object_path, obj = publisher.call_args[0]
+
+        self.assertEqual(task_path, object_path)
+        self.assertIsInstance(obj, TaskInterface)
+
+        self.assertIsInstance(obj.implementation, FCOEDiscoverTask)
+        self.assertEqual(obj.implementation._nic, "eth0")
+        self.assertEqual(obj.implementation._dcb, False)
+        self.assertEqual(obj.implementation._auto_vlan, True)
+
+    @patch('pyanaconda.modules.storage.fcoe.fcoe.fcoe')
+    def reload_module_test(self, fcoe):
+        """Test ReloadModule."""
+        self.fcoe_interface.ReloadModule()
+        fcoe.startup.assert_called_once_with()
+
+    @patch('pyanaconda.modules.storage.fcoe.fcoe.fcoe')
+    def write_configuration_test(self, fcoe):
+        """Test WriteConfiguration."""
+
+        with tempfile.TemporaryDirectory() as root:
+            self.fcoe_interface.WriteConfiguration(root)
+            fcoe.write.assert_called_once_with(root)
+
+
+class FCOETasksTestCase(unittest.TestCase):
+    """Test FCoE tasks."""
+
+    @patch('pyanaconda.modules.storage.fcoe.discover.fcoe')
+    def discovery_fails_test(self, fcoe):
+        """Test the failing discovery task."""
+        fcoe.add_san.return_value = "Fake error message"
+
+        with self.assertRaises(StorageDiscoveryError) as cm:
+            FCOEDiscoverTask("eth0", False, True).run()
```

Comment:
Similar here. It could be nice to set these to variables because they have names.

Reply:
Fixed.